### PR TITLE
Use object.ReferenceEquals when comparing Actions in SocketInput

### PR DESCRIPTION
**Pull request by halter73**

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs [hunk 1]
```diff
@@ -127,8 +127,8 @@ public void IncomingComplete(int count, Exception error)
                 _manualResetEvent.Set();
             }
 
-            if (awaitableState != _awaitableIsCompleted &&
-                awaitableState != _awaitableIsNotCompleted)
+            if (!object.ReferenceEquals(awaitableState, _awaitableIsCompleted) &&
```
Comment (moozzyk) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: Any reason to specify `object.`? you did not do this above...

Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: No. I'll remove `object` from these changes.

**Comment by halter73** [comment]
This is meant to address https://github.com/aspnet/KestrelHttpServer/issues/555

File: src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs (hunk 1 above)
Comment (moozzyk) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: :+1:

**Comment by cesarbs** [comment]
Did you see any perf improvement from this?

**Comment by halter73** [comment]
Frankly, I didn't measure this one. Considering we are doing `Interlocked.CompareExchange` on the `_awaitableState` which is effectively doing a reference comparison too, I felt ReferenceEquals was more correct in addition to generating simpler IL.

**Comment by cesarbs** [comment]
Sounds good. :ship:

**Comment by cesarbs** [comment]
Merge?

**Pull request merged**